src/axom/primal/geometry/NURBSPatch.hpp

```diff
@@ -2088,446 +2414,155 @@ class NURBSPatch
-              }
-            }
+  /*!
+   * \brief Get array of trimming curves
+   */
+  const TrimmingCurveVec& getTrimmingCurves() const { return m_trimmingCurves; }
 
-            if(b < n + p + 1)
-            {
-              for(int row = 0; row <= m; ++row)
-              {
-                strips[ns + 1](save, row) = strips[ns](p, row);
-                if(isRationalPatch)
-                {
-                  strips[ns + 1].setWeight(save, row, strips[ns].getWeight(p, row));
-                }
-              }
-            }
-          }
-        }
+  /// \brief Get mutable array of trimming curves
+  TrimmingCurveVec& getTrimmingCurves() { return m_trimmingCurves; }
 
-        ++ns;
+  /// \brief Get a trimming curve by index
+  const TrimmingCurveType& getTrimmingCurve(int idx) const
+  {
+    SLIC_ASSERT(idx >= 0 && idx < m_trimmingCurves.size());
+    return m_trimmingCurves[idx];
+  }
 
-        if(b < n + p + 1)
-        {
-          for(int j = p - mult; j <= p; ++j)
-          {
-            for(int row = 0; row <= m; ++row)
-            {
-              strips[ns](j, row) = m_controlPoints(b - p + j, row);
-              if(isRationalPatch)
-              {
-                strips[ns].setWeight(j, row, m_weights(b - p + j, row));
-              }
-            }
-          }
-          a = b;
-          b++;
-        }
-      }
-    }
+  /// \brief Add a trimming curve
+  void addTrimmingCurve(const TrimmingCurveType& curve)
+  {
+    m_isTrimmed = true;
+    m_trimmingCurves.push_back(curve);
+  }
 
-    // For each strip, do Bezier extraction on the v-axis
-    axom::Array<BezierPatch<T, NDIMS>> beziers(kp * kq);
-    for(int i = 0; i < beziers.size(); ++i)
+  /// \brief Add array of trimming curves
+  void addTrimmingCurves(const TrimmingCurveVec& curves)
+  {
+    m_isTrimmed = true;
+    for(int i = 0; i < curves.size(); ++i)
     {
-      beziers[i].setOrder(p, q);
-      if(isRationalPatch)
-      {
-        beziers[i].makeRational();
-      }
+      m_trimmingCurves.push_back(curves[i]);
     }
+  }
 
-    for(int s_i = 0; s_i < strips.size(); ++s_i)
-    {
-      auto& strip = strips[s_i];
-      int n_i = strip.getNumControlPoints_u() - 1;
-      int nb = s_i * m_knotvec_v.getNumKnotSpans();
-
-      // Handle this case separately
-      if(q == 0)
-      {
-        for(int i = 0; i < m + 1; ++i)
-        {
-          for(int col = 0; col < n_i + 1; ++col)
-          {
-            beziers[nb](col, 0) = strip(col, i);
-            if(isRationalPatch)
-            {
-              beziers[nb].setWeight(col, 0, strip.getWeight(col, i));
-            }
-          }
+  /// \brief Clear trimming curves, but DON'T mark as untrimmed
+  void clearTrimmingCurves() { m_trimmingCurves.clear(); }
 
-          ++nb;
-        }
+  /// \brief Get number of trimming curves
+  int getNumTrimmingCurves() const { return m_trimmingCurves.size(); }
 
-        continue;
-      }
+  /// \brief use array size as flag for trimmed-ness
```

Comment:
This comment reflects an older implementation of the class, so it should probably be updated to reflect the fact that the flag is the flag for trimmed-ness.
```suggestion
  /// \brief use boolean flag for trimmed-ness
```